rct229/rulesets/ashrae9012019/section11/section11rule7.py

```diff
@@ -0,0 +1,181 @@
+                swh_equip_id: get_swh_equipment_type(rmd_b, swh_equip_id)
+                for swh_equip_id in find_all(
+                    "$.service_water_heating_equipment[*].id", rmd_b
+                )
+            }
+
+            return {
+                "service_water_heating_uses_p": service_water_heating_uses_p,
+                "swh_equip_type_b": swh_equip_type_b,
+            }
+
+        def create_context_list(self, context, data=None):
+            rmd_b = context.BASELINE_0
+            rmd_p = context.PROPOSED
+            is_leap_year_b = data["is_leap_year"]
+
+            swh_bats_and_uses_b = get_swh_components_associated_with_each_swh_bat(
+                rmd_b, is_leap_year_b
+            )
+            swh_bats_and_uses_p = get_swh_bats_and_swh_use(rmd_p)
+
+            building_area_type_SWH_equip_dict = {}
+            building_area_type_and_uses = {}
+            for bat_type, SWH_Equipment_Associations in swh_bats_and_uses_b.items():
+                building_area_type_SWH_equip_dict[bat_type] = {}
+                building_area_type_SWH_equip_dict[bat_type]["id"] = bat_type
+                building_area_type_SWH_equip_dict[bat_type][
+                    "SWH_Equipment_Associations"
+                ] = SWH_Equipment_Associations
+
+                building_area_type_and_uses[bat_type] = {}
+                building_area_type_and_uses[bat_type]["id"] = bat_type
+                building_area_type_and_uses[bat_type][
+                    "swh_bats_and_uses_p"
+                ] = swh_bats_and_uses_p[bat_type]
+
+            return [
+                produce_ruleset_model_description(
+                    USER=False,
+                    BASELINE_0=building_area_type_SWH_equip_dict[bat_type],
+                    PROPOSED=building_area_type_and_uses[bat_type],
+                )
+                for bat_type, SWH_Equipment_Associations in swh_bats_and_uses_b.items()
+            ]
+
+        class SWHBATRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section11Rule7.RMDRule.SWHBATRule, self).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                )
+
+            def is_applicable(self, context, data=None):
+                building_area_type_and_uses_b = context.PROPOSED
+                service_water_heating_uses_p = data["service_water_heating_uses_p"]
+                shw_bat_p = building_area_type_and_uses_b["id"]
+
+                return all(
+                    [
+                        service_water_heating_uses_p[swh_uses_id_p] > 0.0
+                        or shw_bat_p != SERVICE_WATER_HEATING_SPACE.PARKING_GARAGE
+                        for swh_uses_id_p in building_area_type_and_uses_b[
+                            "swh_bats_and_uses_p"
+                        ]
+                    ]
+                )
```

Comment:
This is OK but more fitting to the rule logic is:
```python
return swh_bat_p != SERVICE_WATER_HEATING_SPACE.PARKING_GARAGE or 
all(service_water_heating_uses_p[swh_uses_id_p] > 0.0
for swh_uses_id_p in building_area_type_and_uses_b[
                            "swh_bats_and_uses_p"
                        ])
```

Reply:
Addressed.